pkg/controller/manage/kmesh_manage.go

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kmeshmanage
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"syscall"
+	"time"
+
+	netns "github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	ns "kmesh.net/kmesh/pkg/controller/netns"
+	"kmesh.net/kmesh/pkg/logger"
+)
+
+var (
+	log                = logger.NewLoggerField("kmesh_manage")
+	annotationDelPatch = []byte(fmt.Sprintf(
+		`{"metadata":{"annotations":{"%s":null}}}`,
+		KmeshAnnotation,
+	))
+	annotationAddPatch = []byte(fmt.Sprintf(
+		`{"metadata":{"annotations":{"%s":"%s"}}}`,
+		KmeshAnnotation,
+		"enabled",
+	))
+)
+
+const (
+	DefaultInformerSyncPeriod = 30 * time.Second
+	SpecialIpForKmesh         = "0.0.0.1"
+	EnableKmeshPort           = 929
+	DisableKmeshPort          = 930
+	LabelSelectorKmesh        = "istio.io/dataplane-mode=Kmesh"
+	KmeshAnnotation           = "kmesh.net/redirection"
+)
+
+func NewKmeshManageController(client kubernetes.Interface) error {
+	stopChan := make(chan struct{})
+	nodeName := os.Getenv("NODE_NAME")
+
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(client, DefaultInformerSyncPeriod,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fmt.Sprintf("spec.nodeName=%s", nodeName)
+			options.LabelSelector = LabelSelectorKmesh
+		}))
+
+	informerFactory.Start(wait.NeverStop)
+	informerFactory.WaitForCacheSync(wait.NeverStop)
```

Comment:
This is blocking, so let;s move this to a Run method, so we can run all the controller's Run async without waiting each other.